linkcheck/HtmlParser/htmllib.py

```diff
@@ -186,7 +178,6 @@ def characters (self, data):
         @type data: string
         @return: None
         """
-        data = data.encode(self.encoding, "ignore")
```

Comment:
i agree encoding is an issue. we don't want to start crashing on weird encodings we find out there. and we don't need to be super fancy - from what i can tell from the previous source, we just drop weird chars... in fact, i wonder why those encoding lines are dropped here in the first place... isn't that where we would fix the encoding?